source-mysql/replication.go

```diff
@@ -648,6 +654,27 @@ func (rs *mysqlReplicationStream) tableActive(streamID string) bool {
 	return ok
 }
 
+func (rs *mysqlReplicationStream) schemaActive(schema string) bool {
+	rs.tables.RLock()
+	defer rs.tables.RUnlock()
+	for streamID := range rs.tables.active {
+		var schemaName, _ = splitStreamID(streamID)
+		if strings.EqualFold(schemaName, schema) {
+			return true
+		}
+	}
+	return false
+}
+
+// splitStreamID is the inverse of JoinStreamID and splits a stream name back into
+// separate schema and table name components. Its use is generally kind of a hack
+// and suggests that the surrounding plumbing is probably subtly wrong, because the
+// normalization from schema+table to stream IDs is supposed to go one way only.
+func splitStreamID(streamID string) (string, string) {
```

Comment:
🤔 Go is pretty happy with map keys like:

```
struct StreamID {
   Schema string
   Table string
}
```

Perhaps there's an opportunity to defer this encoding until it's actually needed?

A `StreamID` type could have a method for producing it's `canonical.encoding`.

Reply:
Yeah, that's how we ought to do it at some point. Ideally as part of that refactoring the whole "case normalization" can be dropped in its entirety. It made sense originally, but now that almost everything else is properly case-aware the main thing that normalization does is to make it impossible to capture from tables with "the same name" capitalized differently.